### PR TITLE
Added exception handling to programs data loader

### PR DESCRIPTION
**Pull request by clintonb**
Loading will continue despite a single program failing to load.

ECOM-5078

### CODE REVIEW

**Comment by clintonb** [comment]
FYI @edx/ecommerce

**Comment by clintonb** [comment]
FYI @jibsheet this fixes an error we saw a couple days ago regarding a null value for the marketing_slug field.

**Comment by jibsheet** [comment]
Cool, seems like a good fix, but I'll leave thumbs for your team.

File: course_discovery/apps/course_metadata/data_loaders.py [hunk 1]
```diff
@@ -532,26 +532,32 @@ def ingest(self):
...
+                organization, __ = Organization.objects.get_or_create(
+                    key=org['key'], defaults={'name': org['display_name'], 'partner': self.partner}
+                )
+                organizations.append(organization)
 
-        program.organizations.clear()
-        program.organizations.add(*organizations)
+            program.organizations.clear()
+            program.organizations.add(*organizations)
+        except Exception:  # pylint: disable=broad-except
```
Comment (rlucioni) [inline course_discovery/apps/course_metadata/data_loaders.py:559]: Nitpick: you could use a bare `except` here.

**Comment by rlucioni** [comment]
👍

**Pull request merged**